lib/Overlay/Overlay.js

```diff
@@ -0,0 +1,591 @@
+	kind: Binding,
+
+	/**
+	* @private
+	*/
+	transform: function (overlayComponents) {
+		return updateOwnership(overlayComponents, this.getInstanceOwner());
+	}
+});
+
+/**
+* Adds observers for the known overlay properties which forward the property changes to the
+* contained control with @{link moon.OverlaySupport} identified by
+* {@linke moon.OverlayContainerSupport#overlayTarget}
+*
+* @mixin moon.OverlayContainerSupport
+* @module moonstone/Overlay
+* @public
+*/
+module.exports.Container = {
+
+	/**
+	* @private
+	*/
+	name: 'moon.OverlayContainerSupport',
+
+	/**
+	* Name of the contained control with @{link moon.OverlaySupport}
+	*
+	* @name overlayTarget
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+
+	/**
+	* List of overlay properties to forward to this container's
+	* {@linke moon.OverlayContainerSupport#overlayTarget}
+	*
+	* @private
+	*/
+	overlayPropertyBindings: [
+		'overlayComponents',
+		'overlayShowing',
+		'overlayClasses',
+		'overlayIcon',
+		'overlayTitle',
+		'overlaySubtitle',
+		'overlayPosition',
+		'overlayAlign',
+		'overlayTransparent'
+	],
+
+	/**
+	* @private
+	*/
+	create: kind.inherit(function (sup) {
+		return function () {
+			sup.apply(this, arguments);
+
+			var fn = this.bindSafely(forwardPropertyChange);
+			this.overlayPropertyBindings.forEach(function (p) {
+				this.observe(p, fn);
+				if (utils.exists(this[p])) fn(null, this[p], p);
+			}, this);
+		}
+	})
+};
+
+/**
+* Sets the default configuration and components for selection overlay. Requires that the control to
+* which it is applied also applies {@link moon.OverlaySupport} or contains a control that does and
+* applies {@link moon.OverlayContainerSupport}.
+*
+* SelectionOverlaySupport adds the styling for a centered icon that can be shown when the `selected`
+* class is applied to the overlaid control.
+*
+* ```
+* {kind: Image, src: 'assets/movie.png', mixins: [Overlay.Support, Overlay.Selection],
+* 	overlayShowing: 'spotlight', overlayComponents: [
+*		{kind: Icon, icon: 'check'}
+* 	]
+* }
+* ```
+*
+* @mixin moon.SelectionOverlaySupport
+* @module moonstone/Overlay
+* @public
+*/
+module.exports.Selection = {
+
+	/**
+	* @private
+	*/
+	name: 'moon.SelectionOverlaySupport',
+
+	/**
+	* @private
+	*/
+	overlayClasses: 'moon-overlay-selection',
+
+	/**
+	* @private
+	*/
+	overlayTransparent: true,
+
+	/**
+	* @private
+	*/
+	overlayPosition: 'centered',
+
+	/**
+	* Icon to be displayed when selected. May either be a URL to an image or the valid name of an
+	* {@link moon.Icon#icon}.
+	*
+	* @name overlayIcon
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+
+	/**
+	* @private
+	*/
+	bindings: [
+		// to reduce API surface area, we'll only expose overlayIcon and map that to both src and
+		// icon ensuring that only 1 is valued based on the presence of a `.` to indicate a URL
+		{from: 'overlayIcon', to: '$.overlayIcon.src', transform: function (icon) {
+			return (!icon || icon.indexOf('.') >= 0) ? null : icon;
```

Comment:
This `.` requirement means that icons which reference "http://..." URLs, icons which are targeted using server relative URLs: "/myicon.png", and icon files in the root library directory (referenced with just "lib/badPlaceForAssets/myicon.png", which is technically poor form) will not be supported. This may or may not be a big deal, just something to note which differs from vanilla moon.Icon.